config/snapshot/config_snapshot.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2022 The TestGrid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package snapshot
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/testgrid/config"
+	configpb "github.com/GoogleCloudPlatform/testgrid/pb/config"
+	"github.com/GoogleCloudPlatform/testgrid/util/gcs"
+	"github.com/sirupsen/logrus"
+)
+
+// Config is a mapped representation of the config at a particular point in time
+type Config struct {
+	dashboards map[string]*configpb.Dashboard
+	groups     map[string]*configpb.TestGroup
+	attrs      storage.ReaderObjectAttrs
+	lock       sync.RWMutex
+}
+
+// Dashboard returns a dashboard's configuration
+func (cs *Config) Dashboard(name string) *configpb.Dashboard {
+	if cs == nil {
+		return nil
+	}
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+
+	return cs.dashboards[name]
+}
+
+// AllDashboards returns all dashboards
+func (cs *Config) AllDashboards() []*configpb.Dashboard {
+	if cs == nil {
+		return nil
+	}
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+
+	db := make([]*configpb.Dashboard, len(cs.dashboards))
+	var i int
```

Comment:
Does i do anything here?

Reply:
Yeah, it's used for the index for the loop on the next few lines. This seemed the clearest way to copy the values in a map into a slice.

Reply:
Oh lol, not sure how I missed that.